### PR TITLE
[Merged by Bors] - Add SSL support to kafka-sink 

### PR DESCRIPTION
**Pull request by simlay**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -30,7 +30,9 @@ fluvio             = { git = "https://github.com/infinyon/fluvio", branch = "mas
 fluvio-smartengine = { git = "https://github.com/infinyon/fluvio", branch = "master" }
 fluvio-protocol    = { git = "https://github.com/infinyon/fluvio", branch = "master" }
 fluvio-spu-schema  = { git = "https://github.com/infinyon/fluvio", branch = "master" }
+sasl2-sys          = { git = "https://github.com/infinyon/rust-sasl/", branch = "fix-cross-compiling" }
```
Comment (sehz) [inline Cargo.toml:33]: can we fork instead of depend on branch?

Comment (simlay) [reply Cargo.toml:33]: You mean have it point to `master` or a revision? Yeah, that should be fine.

Comment (morenol) [reply Cargo.toml:33]: Could you create a PR in upstreaming repo to see if they can merge this change

File: Cargo.toml [hunk 2]
```diff
@@ -30,6 +30,8 @@ fluvio             = { git = "https://github.com/infinyon/fluvio", branch = "mas
...
+krb5-src           = { git = "https://github.com/infinyon/rust-krb5-src", branch = "fix-cross-compiling" }
```
Comment (morenol) [inline Cargo.toml:34]: Could you create a PR in upstreaming repo to see if they can merge this change?

File: rust-connectors/sinks/kafka/Cargo.toml [hunk 3]
```diff
@@ -17,5 +17,7 @@ serde = "1"
 tokio-stream = "0.1"
 fluvio-future = { version = "0.4.1", features = ["subscriber"] }
 
-rdkafka = { version = "0.28", features = ["cmake-build"] }
-kafka = "0.9.0"
+# There is some difference between rev v0.26.0 and the 0.26.0 on crates.io.
+# Changing it to use the crates.io version breaks the musl cross compile.
+rdkafka = { git="https://github.com/fede1024/rust-rdkafka", rev="v0.26.0", features = ["cmake-build", "libz-static", "tokio", "ssl-vendored" , "gssapi-vendored"], default-features = false  }
```
Comment (morenol) [inline rust-connectors/sinks/kafka/Cargo.toml:22]: is this still required?

Comment (simlay) [reply rust-connectors/sinks/kafka/Cargo.toml:22]: Yeah. This is still required. https://github.com/fede1024/rust-rdkafka/issues/446#issuecomment-1168043359 it's not ideal but has to do with the openssl runtime.

File: Makefile [hunk 4]
```diff
@@ -29,7 +29,7 @@ smoke-test:
 	$(CARGO_BUILDER) run --bin fluvio-connector start ./test-connector/config.yaml
 
 build:
-	$(CARGO_BUILDER) build $(TARGET_FLAG) $(RELEASE_FLAG) --bin $(CONNECTOR_NAME)
+	$(CARGO_BUILDER) build -vvv $(TARGET_FLAG) $(RELEASE_FLAG) --bin $(CONNECTOR_NAME)
```
Comment (morenol) [inline Makefile:32]: maybe we can remove now the `-vvv`

**Review by morenol — APPROVED** [review]
I think that we need integration tests, maybe we can use docker-compose for a local kafka instance. WIll approve to see if we can merge to test using a docker hub image (unable to build with macos)

**Pull request closed without merging**